terraform/modules/rds-serverless/main.tf

```diff
@@ -0,0 +1,90 @@
+provider "aws" {
+  region = "ap-southeast-2"
+}
```

Comment:
no need to set a provider, the provider of the main code (invoking the module) will be used.